src/common/HTTPRequest.ts

```diff
@@ -66,44 +143,120 @@ export class HTTPRequest {
   }
 
+  /**
+   * @returns the URL of the request
+   */
   url(): string {
     return this._url;
   }
 
+  /**
+   * Contains the request's resource type as it was perceived by the rendering
+   * engine.
+   * @remarks
+   * @returns one of the following: `document`, `stylesheet`, `image`, `media`,
+   * `font`, `script`, `texttrack`, `xhr`, `fetch`, `eventsource`, `websocket`,
+   * `manifest`, `other`.
+   */
   resourceType(): string {
+    // TODO (@jackfranklin): protocol.d.ts has a type for this, but all the
+    // string values are uppercase. The Puppeteer docs explicitly say the
+    // potential values are all lower case, and the constructor takes the event
+    // type and calls toLowerCase() on it, so we can't reuse the type from the
+    // protocol.d.ts. Why do we lower case?
     return this._resourceType;
   }
 
+  /**
+   * @returns the method used (`GET`, `POST`, etc.)
+   */
   method(): string {
     return this._method;
   }
 
+  /**
+   * @returns the request's post body, if any.
+   */
   postData(): string | undefined {
     return this._postData;
   }
 
+  /**
+   * @returns an object with HTTP headers associated with the request. All
+   * header names are lower-case.
+   */
   headers(): Record<string, string> {
     return this._headers;
   }
 
+  /**
+   * @returns the response for this request, if a response has been received.
+   */
   response(): HTTPResponse | null {
     return this._response;
   }
 
+  /**
+   * @returns the frame that initiated the request.
+   */
   frame(): Frame | null {
     return this._frame;
   }
 
+  /**
+   * @returns true if the request is the driver of the current frame's navigation.
+   */
   isNavigationRequest(): boolean {
     return this._isNavigationRequest;
   }
 
+  /**
+   * @remarks
+   *
+   * `redirectChain` is shared between all the requests of the same chain.
+   *
+   * For example, if the website `http://example.com` has a single redirect to
+   * `https://example.com`, then the chain will contain one request:
+   *
+   * ```js
+   * const response = await page.goto('http://example.com');
+   * const chain = response.request().redirectChain();
+   * console.log(chain.length); // 1
+   * console.log(chain[0].url()); // 'http://example.com'
+   * ```
+   *
+   * If the website `https://google.com` has no redirects, then the chain will be empty:
+   *
+   * ```js
+   * const response = await page.goto('https://google.com');
+   * const chain = response.request().redirectChain();
+   * console.log(chain.length); // 0
+   * ```
+   *
+   * @returns the chain of requests - if a server responds with at least a
+   * single redirect, this chain will contain all requests that were redirected.
+   */
   redirectChain(): HTTPRequest[] {
     return this._redirectChain.slice();
   }
 
   /**
-   * @returns {?{errorText: string}}
+   * Access information about the request's failure.
+   *
+   * @remarks
+   *
+   * Example of logging all failed requests:
```

Comment:
Do we want to separate these in a `@example` tag?

Reply:
Yes, I think we should! Good catch, thanks :)